metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_source.py

```diff
@@ -0,0 +1,616 @@
+                            source_platform_name
+                        ].dremio_source_type = self.reference_source_mapping.get_category(
+                            source_type
+                        )
+                        source_map[source_platform_name].root_path = root_path
+                        source_map[source_platform_name].database_name = database_name
+                        source_present = True
+                        break
+
+                if not source_present:
+                    try:
+                        dremio_source_type = self.reference_source_mapping.get_category(
+                            source_type
+                        )
+                    except Exception as exc:
+                        logger.info(
+                            f"Source {source_type} is not a standard Dremio source type. "
+                            f"Adding source_type {source_type} to mapping as database. Error: {exc}"
+                        )
+
+                        self.reference_source_mapping.add_mapping(
+                            source_type, source_name
+                        )
+                        dremio_source_type = self.reference_source_mapping.get_category(
+                            source_type
+                        )
+
+                    source_map[source_platform_name.lower()] = DremioSourceMapping(
+                        platform=source_type,
+                        platform_name=source_name,
+                        dremio_source_type=dremio_source_type,
+                    )
+
+            else:
+                logger.error(
+                    f'Source "{source.container_name}" is broken. Containers will not be created for source.'
+                )
+                logger.error(
+                    f'No new cross-platform lineage will be emitted for source "{source.container_name}".'
+                )
+                logger.error("Fix this source in Dremio to fix this issue.")
+
+        return source_map
+
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+        """
+        Generate workunits for Dremio metadata.
+        """
+
+        self.source_map = self._build_source_map()
+
+        for wu in self.get_workunits_internal():
+            self.report.report_workunit(wu)
+            yield wu
+
+        # Emit the stale entity removal workunits
+        yield from self.stale_entity_removal_handler.gen_removed_entity_workunits()
+
+    def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
+        """
+        Internal method to generate workunits for Dremio metadata.
+        """
+
+        # Process Containers
+        containers = self.dremio_catalog.get_containers()
+        with ThreadPoolExecutor(max_workers=self.max_workers) as executor:
+            future_to_container = {
+                executor.submit(self.process_container, container): container
+                for container in containers
+            }

```

Comment:
Why are we using multithreading here ? I don't see any IO bound call involved in `self.process_container`.

Reply:
Not needed removed